pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/renderer/TextValueProvider.java

```diff
@@ -1,101 +1,224 @@
+                String camelCasedFieldName = fieldName;
+                if (Character.isUpperCase(camelCasedFieldName.charAt(0))) {
+                    camelCasedFieldName = Arrays.stream(fieldName.split("_"))
+                            .map(word -> StringUtils.capitalize(StringUtils.toRootLowerCase(word))).collect(Collectors.joining());
+                    camelCasedFieldName = StringUtils.uncapitalize(camelCasedFieldName);
+                }
 
-        return answer;
+                if (dynamicFieldMap.get(camelCasedFieldName) != null) {
+                    targetMap = dynamicFieldMap;
+                    fieldName = camelCasedFieldName;
+                }
+            }
+        }
+        // if this isn't any of the special cases above, just use the passed-in map
+        if (targetMap == null) {
+            targetMap = moduleMap;
+        }
+        return targetMap.getOrDefault(fieldName, StringUtils.EMPTY);
     }
 
-    protected void removeOptionsFromAnswer(Collection<?> answer, List<Map<String, String>> optionDetails) {
-        List<String> details = optionDetails.stream().map(options -> options.get(ESObjectConstants.OPTION)).collect(Collectors.toList());
-        answer.removeAll(details);
+    protected List<Object> getRawAnswerValues(Map<String, Object> moduleMap, FilterExportConfig filterConfig) {
+        Map<String, Object> answerObject = getRelevantAnswer(moduleMap, filterConfig);
+        Object rawValue = extractValuesFromAnswer(answerObject, filterConfig);
+        if (answerObject != null) {
+            Object optionDetails = answerObject.get(ESObjectConstants.OPTIONDETAILS);
+            if (optionDetails != null && !((List<?>) optionDetails).isEmpty()) {
+                filterConfig.setHasDetails(true);
+            }
+        }
+
+        if (!(rawValue instanceof List)) {
+            return Collections.singletonList(rawValue);
+        }
+        return (List) rawValue;
     }
 
-    protected List<String> getOptionDetails(List<?> optionDetails) {
-        return optionDetails.stream().map(optionDetail ->
-                        new StringBuilder(((Map) optionDetail).get(ESObjectConstants.OPTION).toString())
-                                .append(':')
-                                .append(((Map) optionDetail).get(ESObjectConstants.DETAIL)).toString())
-                .collect(Collectors.toList());
+    protected Map<String, Object> getRelevantAnswer(Map<String, Object> moduleMap, FilterExportConfig filterConfig) {
+        if (moduleMap != null) {
+            List<Map<String, Object>> allAnswers =
+                    (List<Map<String, Object>>) moduleMap.get(ElasticSearchUtil.QUESTIONS_ANSWER);
+            if (allAnswers != null) {
+                return allAnswers.stream()
+                        .filter(ans -> filterConfig.getColumn().getName().equals(ans.get(ESObjectConstants.STABLE_ID)))
+                        .findAny().orElse(null);
+            }
+        }
+        return null;
```

Comment:
I think introducing null object design pattern for this exporting feature could be useful.